packages/@aws-cdk/aws-apigateway/README.md

```diff
@@ -1,2 +1,177 @@
+
+api.root.addMethod('ANY');
+
+const books = api.root.addResource('books');
+books.addMethod('GET');
+books.addMethod('POST');
+
+const book = books.addResource('{book_id}');
+book.addMethod('GET');
+book.addMethod('DELETE');
+```
+
+### Integration Targets
+
+Methods are associated with backend integrations, which are invoked when this
+method is called. API Gateway supports the following integrations:
+
+ * `MockIntegration` - can be used to test APIs. This is the default
+   integration if one is not specified.
+ * `LambdaIntegration` - can be used to invoke an AWS Lambda function.
+ * `AwsIntegration` - can be used to invoke arbitrary AWS service APIs.
+ * `HttpIntegration` - can be used to invoke HTTP endpoints.
+
+The following example shows how to integrate the `GET /book/{book_id}` method to
+an AWS Lambda function:
+
+```ts
+const getBookHandler = new lambda.Function(...);
+const getBookIntegration = new apigateway.LambdaIntegration(getBookHandler);
+book.addMethod('GET', getBookIntegration);
+```
+
+Integration options can be optionally be specified:
+
+```ts
+const getBookIntegration = new apigateway.LambdaIntegration(getBookHandler, {
+  contentHandling: apigateway.ContentHandling.ConvertToText, // convert to base64
+  credentialsPassthrough: true, // use caller identity to invoke the function
+});
+```
+
+Method options can optionally be specified when adding methods:
+
+```ts
+book.addMethod('GET', getBookIntegration, {
+  authorizationType: apigateway.AuthorizationType.IAM,
+  apiKeyRequired: true
+});
+```
+
+#### Default Integration and Method Options
+
+The `defaultIntegration` and `defaultMethodOptions` properties can be used to
+configure a default integration at any resource level. These options will be
+used when defining method under this resource (recursively) with undefined
+integration or options.
+
+> If not defined, the default integration is `MockIntegration`. See reference
+documentation for default method options.
+
+The following example defines the `booksBackend` integration as a default
+integration. This means that all API methods that do not explicitly define an
+integration will be routed to this AWS Lambda function.
+
+```ts
+const booksBackend = new apigateway.LambdaIntegration(...);
+const api = new apigateway.RestApi(this, 'books', {
+  defaultIntegration: booksBackend
+});
+
+const books = new api.root.addResource('books');
+books.addMethod('GET');  // integrated with `booksBackend`
+books.addMethod('POST'); // integrated with `booksBackend`
+
+const book = books.addResource('{book_id}');
+book.addMethod('GET');   // integrated with `booksBackend`
+```
+
+### Deployments
+
+By default, the `RestApi` construct will automatically create an API Gateway
+[Deployment] and a "prod" [Stage] which represent the API configuration you defined in
+your CDK app. This means that when you deploy your app, your API can be accessed
+from the public internet via the stage URL.
```

Comment:
"The public internet" always feels a bit like a pleonasm to me (for "the private internet" feels like an oxymoron).

Reply:
How would you phrase this?

Reply:
Okay, rephrased to "your API will be have open access from the internet via the stage URL.